cfme/tests/cloud/test_tag_mapping.py

```diff
@@ -41,8 +45,36 @@ def refresh_provider(provider):
     return True
 
 
+@pytest.fixture(params=['instances', 'images'])
+def tag_mapping_items(request, appliance, provider):
+    entity_type = request.param
+    collection = getattr(appliance.collections, 'cloud_{}'.format(entity_type))
+    collection.filters = {'provider': provider}
+    view = navigate_to(collection, 'AllForProvider')
+    name = view.entities.get_first_entity().name
+    try:
+        mgmt_item = (
+            provider.mgmt.get_template(name)
+            if entity_type == 'images'
+            else provider.mgmt.get_vm(name)
+        )
+    except ImageNotFoundError:
+        msg = 'Failed looking up template [{}] from CFME on provider: {}'.format(name, provider)
+        logger.exception(msg)
+        pytest.skip(msg)
```

Comment:
Q: Mike small confusion, why we need to log exception finally we are getting `Skipped` exception from `pytest`?

Reply:
I like to use logger.exception because that will capture the actual exception object and log it, along with my contextual message.

Then pytest.skip with the contextual message, so that is what gets displayed in the artifact report for this failure. The log then contains the complete exception object and traceback.